notebook/converted/semi-supervision_feasibility.py

```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Semi-supervision Feasibility
+# This notebook evaluates the data to see how much unlabeled blood data exists in the refine.bio human compendium. If enough exists, it will be important to evaluate whether semi-supervision helps model performance
+
+# In[1]:
+
+
+import collections
+import json
+import os
+import pickle
+
+import pandas as pd
+
+from whistl import utils
+
+
+# In[2]:
+
+
+data_dir = '../../data/'
+map_file = os.path.join(data_dir, 'sample_classifications.pkl')
+
+sample_to_label = utils.parse_map_file(map_file)
+with open(map_file, 'rb') as in_file:
+    label_to_sample = pickle.load(in_file)[0]
+
+
+# In[3]:
+
+
+metadata_path = os.path.join(data_dir, 'human_compendium/aggregated_metadata.json')
+with open(metadata_path) as json_file:
+    metadata = json.load(json_file)
+
+
+# In[4]:
+
+
+def get_tissue(sample_metadata, sample):
+    '''Take a sample as input and return the tissue if that information is 
+       present, otherwise return None
+    '''
+    try:
+        characteristics = sample_metadata[sample]['refinebio_annotations'][0]['characteristics_ch1']
+        for characteristic in characteristics:
+            if 'tissue:' in characteristic:
+                tissue = characteristic.split(':')[1]
+                tissue = tissue.strip().lower()
+                return tissue
+            
+    # Catch exceptions caused by a field not being present
+    except KeyError:
+        return None
+    
+    # 'refinebio_annotations' is usually a length 1 list containing a dictionary.
+    # Sometimes it's a length 0 list indicating there aren't annotations
+    except IndexError:
+        return None
+
+
+# In[5]:
+
+
+sample_metadata = metadata['samples']
+
+tissues = []
+for sample in sample_metadata:
+    tissue = get_tissue(sample_metadata, sample)
+    if tissue is not None:
+        tissues.append(tissue)
+
+
+# In[6]:
+
+
+tissue_counts = collections.Counter(tissues)
+tissue_counts.most_common()[:5]
+
+
+# In[7]:
+
+
+keys = tissue_counts.keys()
+blood_keys = []
+for key in keys:
+    if 'blood' in key or 'pbmc' in key:
+        blood_keys.append(key)
+sorted(blood_keys)
```

Comment:
That's a good question. I'm interested in blood since it's arguably the easiest tissue to sample for the purpose of diagnosing diseases.

It's a little unclear what you mean by 'different sources of blood samples'. If you mean between individuals, that's partially what this project will determine. If you mean whole blood vs. pbmcs, that's an analysis that I will need to run to make sure.